### ISSUE TITLE
Bugs in highlighting variables in anonymous functions

### ISSUE THREAD

**Issue by przepompownia** [phpactor/phpactor#1646]
Let's look at self-explained examples:
![image](https://user-images.githubusercontent.com/11404453/170129828-ea49d6ab-50d4-4a7a-8c50-d58932a06616.png)
![image](https://user-images.githubusercontent.com/11404453/170129907-79e8f9eb-35e5-45a8-8e13-cea1c86b276c.png)
![image](https://user-images.githubusercontent.com/11404453/170130094-a272880f-cd3f-41a3-8768-833fff1310b0.png)

**Comment by dantleech** [comment]
i guess you are suggesting they it should respect scope?

currently it will highlight all variables named `$foo` in the entire document

**Comment by przepompownia** [comment]
Yes - it's about scope. Nothing especially annoying - I added these minor bugs to remember about them.

**Comment by mamazu** [comment]
Where would that logic be located? I'd like to help with this but don't know where to look.

**Comment by dantleech** [comment]
code is here: https://github.com/phpactor/phpactor/blob/3dadcf86a19dc6c16c8ceca2cc9c0be62867f874/lib/Extension/LanguageServerReferenceFinder/Model/Highlighter.php#L27

**Comment by przepompownia** [comment]
While I'm not ready to investigate how to fix it, let me append other cases here.

- scope of virtual variables:
![image](https://user-images.githubusercontent.com/11404453/173570324-9500a791-311d-4dbb-ba46-fcaee81df56c.png)
- promoted properties:
![image](https://user-images.githubusercontent.com/11404453/173571427-d3378bc0-655c-4def-92cd-c80c137f0f0e.png)

**Comment by przepompownia** [comment]
With https://github.com/phpactor/phpactor/compare/master...przepompownia:highlight-promoted-properties?expand=1 and still without respecting the current scope I got one-directional highlighnig (i.e. if the cursor is at the place of access, the definition gets highlighted and not vice versa at the moment):
![image](https://user-images.githubusercontent.com/11404453/173697203-a0823873-e659-499b-aa9e-162ab3bc686b.png)
